### PR TITLE
Parse the closest class for functions and properties

### PR DESCRIPTION
**Pull request by sarahlim**
Associated Issue: #4021

### Summary of Changes

- parsing function symbols now includes the name of the closest ancestor class, or `null` if there is no containing class
- can be used like this:

```diff
diff --git a/src/components/PrimaryPanes/Outline.js b/src/components/PrimaryPanes/Outline.js
index 5c0fac6..97359e6 100644
--- a/src/components/PrimaryPanes/Outline.js
+++ b/src/components/PrimaryPanes/Outline.js
@@ -45,7 +45,7 @@ export class Outline extends Component {
   }

   renderFunction(func: SymbolDeclaration) {
-    const { name, location, parameterNames } = func;
+    const { name, location, parameterNames, klass } = func;

     return (
       <li
@@ -54,6 +54,8 @@ export class Outline extends Component {
         onClick={() => this.selectItem(location)}
       >
         <PreviewFunction func={{ name, parameterNames }} />
+        {" - "}
+        {klass}
       </li>
     );
   }
```

screen shot 2017-09-18 at 12 30 54 pm

### Test Plan

- added component test fixture (should I add more edge cases?)

### CODE REVIEW

File: src/utils/parser/tests/getSymbols.spec.js [hunk 1]
```diff
@@ -53,4 +53,9 @@ describe("Parser.getSymbols", () => {
     const symbols = formatSymbols(getSource("parseScriptTags", "html"));
     expect(symbols).toMatchSnapshot();
   });
+
+  it("component", () => {
+    const symbols = formatSymbols(getSource("component"));
+    expect(symbols).toMatchSnapshot();
+  });
```
Comment (jasonLaster) [inline src/utils/parser/tests/getSymbols.spec.js]: lets update `formatSymbols` to show the class name when it exists, this will get us some additional test coverage.

File: src/utils/parser/getSymbols.js [hunk 2]
```diff
@@ -359,7 +371,8 @@ export function formatSymbols(source: Source) {
       ? symbol.parameterNames.join(", ")
       : "";
     const expression = symbol.expression || "";
-    return `${loc} ${exprLoc} ${expression} ${symbol.name} ${params}`;
+    const klass = symbol.klass || "";
+    return `${loc} ${exprLoc} ${expression} ${symbol.name} ${params} ${klass}`;
```
Comment (jasonLaster) [inline src/utils/parser/getSymbols.js]: hmm, looks like we're not including functions, which we probably should... perhaps we should update `formatSymbol` to map the entire dict of symbols:

```js
Object.keys(symbols).map(name => `${name}:\n ${symbols[name].map(summarize)}`
```

**Comment by jasonLaster** [comment]
Thanks so much!

File: src/utils/parser/getSymbols.js [hunk 3]
```diff
@@ -76,9 +76,8 @@ function getClassName(path: NodePath): ?string {
   if (!classDeclaration) {
     return null;
   }
-  if (classDeclaration) {
-    return classDeclaration.node.id.name;
-  }
+
```
Comment (sarahlim) [inline src/utils/parser/getSymbols.js]: Whoa.......not sure how this happened, thanks for the catch 😛
